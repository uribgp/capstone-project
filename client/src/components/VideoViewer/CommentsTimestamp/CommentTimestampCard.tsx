import React, { ReactElement } from "react";
import { IoMdArrowDown, IoMdArrowUp } from "react-icons/io";
import TitleSmall from "../../Shared/Typography/TitleSmall";
import Button from "../../Shared/Button/Button";
import ButtonIcon from "../../Shared/Button/ButtonIcon";
import UserProfile from "../../Shared/UserProfile/UserProfile";
import "./comment-timestamp.style.scss";
import UpvoteDownvote from "../../Shared/UpvoteDownvote/UpvoteDownvote";
import { CommentVote } from "../../../types/comment-vote";
interface Props {
  comment: string;
  username: string;
  timestamp: number;
  isToggled: boolean;
  onClick: (commentId: number, timestamp: number) => void;
  avatar: string;
  hasVerifiedIcon: boolean;
  id: number;
  formatted_timestamp: string;
  onUpvoteClick: (id: number, voteStatus: CommentVote) => void;
  onDownvoteClick: (id: number, voteStatus: CommentVote) => void;
  score: number | string;
  voteStatus: CommentVote;
}

export default function CommentTimestampCard({
  comment,
  username,
  formatted_timestamp,
  isToggled,
  onClick,
  id,
  hasVerifiedIcon,
  avatar,
  score,
  timestamp,
  onUpvoteClick,
  onDownvoteClick,
  voteStatus,
}: Props): ReactElement {
  const isFaded = () => {
    const faded = { opacity: 0.2 };
    const fullColor = { opacity: 1 };
    return isToggled ? fullColor : faded;
  };

  return (
    <div
      onClick={() => onClick(id, timestamp)}
      style={isFaded()}
      className="comment-timestamp"
    >
      <div className="comment-timestamp--info">
        <div className="comment-timestamp--user-info">
          <UserProfile className="comment-timestamp--user-profile" size="medium" src={avatar} alt="user avatar" />
          <div>
            <div className="comment-timestamp-username">{username}</div>
            <div className="comment-timestamp-timestamp">
              {formatted_timestamp}
            </div>
          </div>
        </div>

        <UpvoteDownvote
          onUpvoteClick={() => onUpvoteClick(id, voteStatus)}
          voteStatus={voteStatus}
          onDownvoteClick={() => onDownvoteClick(id, voteStatus)}
          score={score}
        />
      </div>
      {isToggled && <div className="comment-timestamp-comment">{comment}</div>}
    </div>
  );
}
