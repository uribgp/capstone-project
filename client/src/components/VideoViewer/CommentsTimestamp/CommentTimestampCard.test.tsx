import React, { ReactElement } from "react";
import { shallow, mount, render } from "enzyme";
import CommentTimestampCard from "./CommentTimestampCard";

it("expect to render card component.", () => {
  expect(
    shallow(
      <CommentTimestampCard
        username="pontus"
        id={4}
        onDownvoteClick={() => null}
        voteStatus="upvoted"
        timestamp={15}
        isToggled={false}
        avatar="hej.jpg"
        onClick={() => null}
        onUpvoteClick={() => null}
        hasVerifiedIcon={false}
        comment="Nice lift"
        formatted_timestamp="00:15"
        score={0}
      />
    )
  ).toMatchSnapshot();
});
