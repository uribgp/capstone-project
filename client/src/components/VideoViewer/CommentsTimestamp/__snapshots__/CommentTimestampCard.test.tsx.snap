// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render card component. 1`] = `
<div
  className="comment-timestamp"
  onClick={[Function]}
  style={
    Object {
      "opacity": 0.2,
    }
  }
>
  <div
    className="comment-timestamp-user-wrap"
  >
    <UserProfile
      alt="user avatar"
      src="hej.jpg"
    />
    <div
      className="comment-timestamp-username-timestamp-wrap"
    >
      <TextSmall
        className="comment-timestamp-username"
        text="pontus"
      />
      <div
        className="comment-timestamp-timestamp"
      >
        00:15
      </div>
      <UpvoteDownvote
        onDownvoteClick={[Function]}
        onUpvoteClick={[Function]}
        score={0}
        voteStatus="upvoted"
      />
    </div>
  </div>
</div>
`;
